src/main/java/de/retest/recheck/ignore/Filter.java

```diff
@@ -0,0 +1,58 @@
+package de.retest.recheck.ignore;
+
+import de.retest.recheck.ui.descriptors.Element;
+import de.retest.recheck.ui.descriptors.IdentifyingAttributes;
+import de.retest.recheck.ui.diff.AttributeDifference;
+
+/**
+ * General interface to filter changes during Diffing
+ *
+ * In principal, we create a diff for everything that differs, and put that into the result file. Then it is up to the
+ * UI (GUI or CLI) to hide filtered differences – this is what this interface is for. This is pretty much inline with
+ * how Git works – the diff is there, it just doesn't show up when filtered.
+ *
+ * Can be implemented by reading a file, or by an implementation delivered by the user.
+ */
+public interface Filter {
```

Comment:
The Filter does not define positive or negative filtering, i.e. wether only matching elements are shown or only elements that do not match.